### PR TITLE
Struct SparseValue && Bug Fix

### PR DESCRIPTION
**Pull request by seiriosPlus**
### PR types

New features
### PR changes

Others
### Describe

1.修复在分布式预测时，在使用准入策略是，由于未针对预测场景做针对性过滤触发准入条件导致计数及初始化错误.

2.修复在分布式训练时，由于远端拉取稀疏特征时的去重导致Server端对特征的频率计数错误.

--------

Base speed:
{'speed': [83504, 81608, 81701, 83903, 86251]}
PR speed:
'speed': [82247, 85332, 85383, 83559, 86213]

------

Base AUC:
{'0': 0.7053, '1': 0.7394,  '2': 0.7534, '3': 0.7623, '4': 0.7693}}
PR AUC:
{'0': 0.7112,  '1': 0.7401,  '2': 0.7561,  '3': 0.7639,  '4': 0.7701}

### CODE REVIEW

File: paddle/fluid/distributed/table/depends/sparse_utils.h [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    feasigns_ = reinterpret_cast<uint64_t*>(begin + sizeof(bool));
+    frequencies_ = reinterpret_cast<uint32_t*>(begin + sizeof(bool) +
+                                               sizeof(uint64_t) * numel_);
+  }
+
+  void Fission(const int shard_id, const int shard_num,
+               std::vector<int>* offset_shard) const {
+    offset_shard->reserve(numel_ / shard_num + 1);
+    for (int x = 0; x < numel_; ++x) {
+      if (x % shard_num == shard_id) {
```
Comment (a6802739) [inline paddle/fluid/distributed/table/depends/sparse_utils.h]: 这样写会有问题，应该是 feasigns_[x] % shard_num == shard_id

**Review by luotao1 — DISMISSED** [review]
LGTM for const_cast

**Review by MrChengmo — DISMISSED** [review]
LGTM

**Review by luotao1 — DISMISSED** [review]
LGTM for const_cast

**Review by MrChengmo — DISMISSED** [review]
LGTM

File: paddle/fluid/distributed/fleet.h [hunk 2]
```diff
@@ -84,19 +84,11 @@ class FleetWrapper {
...
-      const Scope& scope, const uint64_t table_id,
-      const std::vector<std::string>& var_names,
-      std::vector<uint64_t>* fea_keys,
-      std::vector<std::vector<float>>* fea_values, int fea_dim);
-
   // Pull sparse variables from server in sync mode
   // pull immediately to tensors
   void PullSparseToTensorSync(const uint64_t table_id, int fea_dim,
                               uint64_t padding_id, platform::Place place,
+                              bool is_training,
```
Comment (raindrops2sea) [inline paddle/fluid/distributed/fleet.h:94]: add comments please.

**Review by raindrops2sea — APPROVED** [review]

**Review by MrChengmo — APPROVED** [review]
LGTM

**Review by luotao1 — APPROVED** [review]

**Review by chalsliu — APPROVED** [review]

**Pull request merged**